# Assignment Day 16-17

Ali Mohammed Taresh Ali 

ali713626792@gmail.com

In [39]:
%%HTML
<style type="text/css">
table.dataframe td, table.dataframe th {
    border-style: solid;
}

In [7]:
%matplotlib inline
import pandas as pd
import numpy as np

In [10]:
titles = pd.read_csv('C:/Users/(pandas 1)/Day 16 - 17 Assingment/Dataset for Pandas Assinment/titles.csv')
print(titles.shape)
titles.head()

(244914, 2)


title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

In [12]:
cast = pd.read_csv('C:/Users/(pandas 1)/Day 16 - 17 Assingment/Dataset for Pandas Assinment/cast.csv')
print(cast.shape)
cast.head()

(3786176, 6)


title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4  31.0
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

### How many movies are listed in the titles dataframe?

In [15]:
len(titles)

244914

In [16]:
titles.count()

title    244914
year     244914
dtype: int64

There are 244914 movies listed in the titles dataframe.

### What are the earliest two films listed in the titles dataframe?

In [17]:
titles.sort_values('year').head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

### How many movies have the title "Hamlet"?

In [19]:
len(titles[titles.title == 'Hamlet'])

20

In [20]:
titles[titles['title'] == "Hamlet"].title.count()

20

There are 20 movies have a title "Hamlet" in dataset.

### How many movies are titled "North by Northwest"?

In [21]:
len(titles[titles.title == "North by Northwest"])

1

There is 1 movie have a title "North by Northwest" in dataset.

### When was the first movie titled "Hamlet" made?

In [22]:
titles[titles.title == "Hamlet"].sort_values('year').head(1)

title  year
52521  Hamlet  1910

In 1910

### List all of the "Treasure Island" movies from earliest to most recent.

In [27]:
titles[titles.title == "Treasure Island"].sort_values('year')

title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [28]:
titles[titles['year'] == 1950].year.count()

1113

There are 1,113 movies made in year 1950.

### How many movies were made in the year 1960?

In [29]:
titles[titles['year'] == 1960].year.count()

1549

There are 1,549 movies made in year 1960.

### How many movies were made from 1950 through 1959?

In [13]:
tm = 0
for i in range(1950, 1960):
    tm += titles[titles['year'] == i].count().year
print(tm)

12934


In [34]:
titles[(titles['year']>=1950) & (titles['year']<=1959)].year.count()

12934

12,934 movies made from 1950 through 1959.

### In what years has a movie titled "Batman" been released?

In [35]:
titles[titles.title == "Batman"]

title  year
39349   Batman  1943
240896  Batman  1989

In 1943 and 1989 movie Batman has been released.

### How many roles were there in the movie "Inception"?

In [38]:
cast

title  year                         name  \
0                          Closet Monster  2015                     Buffy #1   
1                         Suuri illusioni  1985                       Homo $   
2                     Battle of the Sexes  2017                      $hutter   
3                    Secret in Their Eyes  2015                      $hutter   
4                              Steve Jobs  2015                      $hutter   
...                                   ...   ...                          ...   
3786171                           Foxtrot  1988            Lilja ??risd?ttir   
3786172  Niceland (Population. 1.000.002)  2004    Sigr??ur J?na ??risd?ttir   
3786173                         Skammdegi  1985           Dalla ??r?ard?ttir   
3786174                      U.S.S.S.S...  2003  Krist?n Andrea ??r?ard?ttir   
3786175                 Bye Bye Blue Bird  1999                 Rosa ? R?gvu   

            type                character     n  
0          actor                  Buffy 4  31.0  
1          actor                   Guests  22.0  
2          actor          Bobby Riggs Fan  10.0  
3          actor          2002 Dodger Fan   NaN  
4          actor  1988 Opera House Patron   NaN  
...          ...                      ...   ...  
3786171  actress                     D?ra  24.0  
3786172  actress             Woman in Bus  26.0  
3786173  actress            Hj?krunarkona   9.0  
3786174  actress   Afgr.dama ? bens?nst??  17.0  
3786175  actress        Pensionatv?rtinde   NaN  

[3786176 rows x 6 columns]

In [36]:
cast[cast['title']=='Inception'].character.shape[0]

78

In [37]:
cast[cast['title']=='Inception'].character.count()

78

There are 78 roles were there in the movie "Inception".

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [40]:
cast[cast['title']=='Inception'].isnull().sum().sum()

27

There are 27 roles in the movie "Inception" are NOT ranked by an "n" value.

### But how many roles in the movie "Inception" did receive an "n" value?

In [43]:
Inception_Films = cast[cast.title == 'Inception'].notna().n.sum()
Inception_Films

51

51 roles in the movie "Inception" receive an "n" value.

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [45]:
North_By_NorthWest_Films = cast[cast.title == "North by Northwest"]

In [46]:
North_By_NorthWest_Films[North_By_NorthWest_Films.n.notna()].sort_values('n')

title  year                  name     type  \
871426   North by Northwest  1959            Cary Grant    actor   
3518333  North by Northwest  1959       Eva Marie Saint  actress   
1457165  North by Northwest  1959           James Mason    actor   
3157232  North by Northwest  1959   Jessie Royce Landis  actress   
356707   North by Northwest  1959        Leo G. Carroll    actor   
3048272  North by Northwest  1959  Josephine Hutchinson  actress   
1695173  North by Northwest  1959           Philip Ober    actor   
1275337  North by Northwest  1959         Martin Landau    actor   
2447081  North by Northwest  1959         Adam Williams    actor   
1811387  North by Northwest  1959          Edward Platt    actor   
665768   North by Northwest  1959     Robert Ellenstein    actor   
2295714  North by Northwest  1959          Les Tremayne    actor   
464736   North by Northwest  1959       Philip Coolidge    actor   
1508344  North by Northwest  1959         Patrick McVey    actor   
206471   North by Northwest  1959          Edward Binns    actor   
1384780  North by Northwest  1959             Ken Lynch    actor   

                           character     n  
871426            Roger O. Thornhill   1.0  
3518333                  Eve Kendall   2.0  
1457165              Phillip Vandamm   3.0  
3157232              Clara Thornhill   4.0  
356707                 The Professor   5.0  
3048272                Mrs. Townsend   6.0  
1695173              Lester Townsend   7.0  
1275337                      Leonard   8.0  
2447081                     Valerian   9.0  
1811387              Victor Larrabee  10.0  
665768                         Licht  11.0  
2295714                   Auctioneer  12.0  
464736                     Dr. Cross  13.0  
1508344               Sergeant Flamm  14.0  
206471                Captain Junket  15.0  
1384780  Charley - Chicago Policeman  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [47]:
Sleuth = cast[cast.title == 'Sleuth']
Sleuth_1972 = Sleuth[Sleuth.year == 1972].sort_values('n')
print('The entire cast, in "n"-order, of the 1972 film "Sleuth":')
Sleuth_1972

The entire cast, in "n"-order, of the 1972 film "Sleuth":


title  year                name     type  \
1705394  Sleuth  1972    Laurence Olivier    actor   
326773   Sleuth  1972       Michael Caine    actor   
373876   Sleuth  1972      Alec Cawthorne    actor   
1465483  Sleuth  1972  John (II) Matthews    actor   
2724553  Sleuth  1972  Eve (III) Channing  actress   
1448690  Sleuth  1972        Teddy Martin    actor   

                          character    n  
1705394                 Andrew Wyke  1.0  
326773                  Milo Tindle  2.0  
373876            Inspector Doppler  3.0  
1465483  Detective Sergeant Tarrant  4.0  
2724553             Marguerite Wyke  5.0  
1448690      Police Constable Higgs  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [48]:
Sleuth_2007 = Sleuth[Sleuth.year == 2007].sort_values('n')
Sleuth_2007

title  year                 name     type          character    n
326774   Sleuth  2007        Michael Caine    actor             Andrew  1.0
1293634  Sleuth  2007             Jude Law    actor               Milo  2.0
1805907  Sleuth  2007        Harold Pinter    actor        Man on T.V.  3.0
260242   Sleuth  2007      Kenneth Branagh    actor  Other Man on T.V.  NaN
373877   Sleuth  2007  Alec (II) Cawthorne    actor  Inspector Doppler  NaN
2724552  Sleuth  2007    Eve (II) Channing  actress    Marguerite Wyke  NaN
3370424  Sleuth  2007    Carmel O'Sullivan  actress             Maggie  NaN

### How many roles were credited in the silent 1921 version of Hamlet?

In [22]:
cast[(cast['title']=='Hamlet') & (cast['year']==1921)].character.shape[0]

9

In [49]:
cast[(cast['title']=='Hamlet') & (cast['year']==1921)].character.count()

9

9 roles were credited in the silent 1921 version of Hamlet

### How many roles were credited in Branagh’s 1996 Hamlet?

In [50]:
cast[(cast['title']=='Hamlet') & (cast['year']==1996)].character.shape[0]

54

54 roles were credited in Branagh’s 1996 Hamlet

### How many "Hamlet" roles have been listed in all film credits through history?

In [51]:
cast[cast['title']=='Hamlet'].character.shape[0]

342

342 "Hamlet" roles have been listed in all film credits through history.

### How many people have played an "Ophelia"?

In [52]:
cast[cast.character == "Ophelia"].name.count()

117

117 people have played an "Ophelia".

In [54]:
cast[cast.character == "Ophelia"].name.drop_duplicates().count()

115

115 people without duplicates have played an "Ophelia".

### How many people have played a role called "The Dude"?

In [55]:
cast[cast.character == "The Dude"].name.count()

19

19 people have played a role called "The Dude"

### How many people have played a role called "The Stranger"?

In [56]:
cast[cast.character == "The Stranger"].name.count()

212

212 people with duplicates have played a role called "The Stranger".

In [58]:
cast[cast.character == "The Stranger"].name.drop_duplicates().count()

202

202 people without duplicates have played a role called "The Stranger".

### How many roles has Sidney Poitier played throughout his career?

In [73]:
cast[cast.name == "Sidney Poitier"].name.count()

43

In [74]:
cast[cast.name == "Sidney Poitier"].character.count()

43

Sidney Poitier has played 43 roles throughout his career.

### How many roles has Judi Dench played?

In [75]:
cast[cast.name == "Judi Dench"].name.count()

55

Judi Dench has played 55 roles throughout his career.

In [81]:
cast[cast.name == "Judi Dench"].character.drop_duplicates().count()

46

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [82]:
cast[(cast['name'] == 'Cary Grant') & (cast['n'] == 2) & (cast['year'] < 1950) & (cast['year'] >= 1940)].sort_values(by='year')

title  year        name   type    character    n
871423  My Favorite Wife  1940  Cary Grant  actor   Nick Arden  2.0
871433    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [83]:
cast[(cast['name'] == 'Cary Grant') & (cast['n'] == 1) & (cast['year'] < 1950) & (cast['year'] >= 1940)].sort_values(by='year')

title  year        name   type  \
871448               The Howards of Virginia  1940  Cary Grant  actor   
871407                       His Girl Friday  1940  Cary Grant  actor   
871450                The Philadelphia Story  1940  Cary Grant  actor   
871438                             Suspicion  1941  Cary Grant  actor   
871452                  The Talk of the Town  1942  Cary Grant  actor   
871429                 Once Upon a Honeymoon  1942  Cary Grant  actor   
871398                     Destination Tokyo  1943  Cary Grant  actor   
871422                             Mr. Lucky  1943  Cary Grant  actor   
871430                      Once Upon a Time  1944  Cary Grant  actor   
871390                  Arsenic and Old Lace  1944  Cary Grant  actor   
871425             None But the Lonely Heart  1944  Cary Grant  actor   
871424                         Night and Day  1946  Cary Grant  actor   
871427                             Notorious  1946  Cary Grant  actor   
871444      The Bachelor and the Bobby-Soxer  1947  Cary Grant  actor   
871445                     The Bishop's Wife  1947  Cary Grant  actor   
871421  Mr. Blandings Builds His Dream House  1948  Cary Grant  actor   
871402          Every Girl Should Be Married  1948  Cary Grant  actor   
871411                I Was a Male War Bride  1949  Cary Grant  actor   

                                   character    n  
871448                           Matt Howard  1.0  
871407                          Walter Burns  1.0  
871450                    C. K. Dexter Haven  1.0  
871438                      Johnnie Aysgarth  1.0  
871452                          Leopold Dilg  1.0  
871429                 Patrick 'Pat' O'Toole  1.0  
871398                         Capt. Cassidy  1.0  
871422  Joe Adams -posing as Joe Bascopolous  1.0  
871430                           Jerry Flynn  1.0  
871390                     Mortimer Brewster  1.0  
871425                            Ernie Mott  1.0  
871424                           Cole Porter  1.0  
871427                                Devlin  1.0  
871444                           Dick Nugent  1.0  
871445                                Dudley  1.0  
871421                         Jim Blandings  1.0  
871402                     Dr. Madison Brown  1.0  
871411                 Captain Henri Rochard  1.0

### How many roles were available for actors in the 1950s?

In [84]:
cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actor')]['character'].count()

157738

157,738 roles were available for actors in the 1950s

### How many roles were available for actresses in the 1950s?

In [85]:
cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actress')]['character'].count()

57733

There are 57,733 roles available for actresses in 1950's.

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [86]:
cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] == 1)]['character'].count()

65140

65,140 leading roles were available from the beginning of film history through 1980

### How many non-leading roles were available through from the beginning of film history through 1980?

In [87]:
cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] > 1)]['character'].count()

669320

669,320 non-leading roles were available through from the beginning of film history through 1980.

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [88]:
cast[(cast['year'] >= 1894) & (cast['year'] <= 1980)]['n'].isnull().sum()

448347

448,347 roles through 1980 were minor enough that they did not warrant a numeric "n" rank.

In [1]:
import os 
os.getcwd()

'C:\\Users\\unkno\\Desktop\\LetsUpgrade\\LetsUpgrade AI and ML\\Day 17  Exploratory Data Analytics _4(pandas 2)\\Day 16 - 17 Assingment\\Day 16 - 17 Assignment'